pinot-broker/src/main/java/org/apache/pinot/broker/api/HttpRequesterIdentity.java

```diff
@@ -43,4 +45,31 @@ public String getEndpointUrl() {
   public void setEndpointUrl(String endpointUrl) {
     _endpointUrl = endpointUrl;
   }
+
+  @Override
+  // If reverse proxy is used X-Forwarded-For will be populated
+  // If X-Forwarded-For is not present, check if x-real-ip is present
+  // Since X-Forwarded-For can contain comma separated list of values, we convert it to ";" delimiter to avoid
+  // downstream parsing errors for other fields where "," is being used
```

Comment:
```suggestion
  /**
   * If reverse proxy is used X-Forwarded-For will be populated
   * If X-Forwarded-For is not present, check if x-real-ip is present
   * Since X-Forwarded-For can contain comma separated list of values, we convert it to ";" delimiter to avoid
   * downstream parsing errors for other fields where "," is being used
   */
  @Override
```